nx/lib/nx.ex

```diff
@@ -360,6 +360,8 @@ defmodule Nx do
 
   @non_finite [:neg_infinity, :infinity, :nan]
 
+  defguard is_tensor(t) when is_number(t) or is_struct(t, T) or is_struct(t, Complex)
```

Comment:
Let's doc it too! And we need to add a `@doc type: ...` thing. But no idea what to add here. Maybe its own called `:guard`?

Reply:
![Screenshot from 2022-05-18 13-33-32](https://user-images.githubusercontent.com/16843419/169095257-dab2e00c-3385-4d87-9708-171325d17090.png)
![Screenshot from 2022-05-18 13-33-37](https://user-images.githubusercontent.com/16843419/169095263-0e0c92a9-e5ac-4b0c-a818-684406abe5f2.png)

I added it like this. Please take another look and merge if it makes sense!
I've added it as `type: :guards`, should it be `:guard` instead?